### PR TITLE
Feature/listens to

### PR DESCRIPTION
**Pull request by WillzZz**
Could use some feedback on this feature.

This is meant to reduce boilerplate from creation of mediators.

<b>This:</b>
[Inject]
public MySignal MySignal { get; set; }
[Inject]
public MyOtherSignal MyOtherSignal { get; set; }

public override OnRegister()
{
    MySignal.AddListener(MyCallback);
    MyOtherSignal.AddListener(MyOtherCallback);
}

//callbacks
public override OnRemove()
{
    MySignal.RemoveListener(MyCallback);
    MyOtherSIgnal.RemoveListener(MyCallback);
} 

<b>Becomes:</b>
[ListensTo(typeof(MySignal)]
public void MyListener()

[ListensTo(typeof(MyPayloadSignal>)]
public void MyPayloadListener(float r, float g, float b)

<b>Required in context: </b>
injector.Unbind`<IMediationBinder>`();
injector.Bind`<IMediationBinder>`().To`<SignalMediationBinder>`().ToSingleton();

Keep in mind <b> your signals must already be bound</b> It is not the domain of this extension to auto inject those signals. And an [Implements] class annotation takes two seconds. This is just meant to alleviate some of your time spent writing stale mediator code.

Re: order of operations: This is adding and removing listeners before OnRegister and before OnRemove.

Be forewarned! This is potentially slow! I think I'll create an ISignalMediator interface and check for that to avoid unnecessary reflection. Reflection is <b>slow</b>, so it is not recommended to use this for objects which are being created/destroyed regularly! I may cache some more values, as well. The signal values should be cached and I can write a few no-brainer optimizations in loops and property lookups I'm sure. I guess my point is use some common sense :)

### CODE REVIEW

**Comment by fogAndWhisky** [comment]
Spending a bit of time with your PR....

- The first notable thing is that I'm getting a UT failure.
It looks as if the ReflectionBinder itself is getting counted as a reflected item during pre-reflection. Hardly the worst error in the world, to be sure, but should fix.
- There's also stuff to do with named constructor Injection, which is on master, but seemingly is removed here? Why is that?
- The failing test I mentioned above is probably failing because of this line in InjectionBinder:

    Bind<IReflectionBinder>().ToValue(injector.reflector);

- Aside from breaking the test, I think the binding and injecting of the Reflector is unnecessary and unwise. Unnecessary because you could just inject InjectionBinder and access it as InjectionBinder.injector.reflector, and unwise because it's an inducement to start injecting it elsewhere as if it's a thing regular devs should have access to.

I think that covers my list of major concerns. In general, it looks good. I'd suggest you study your diff with master (since the named ctr thing worries me a bit). I'd also look for any ways to optimize, esp in the ReflectionBinder, since that's such a crucial piece of Strange's offering.

**Comment by WillzZz** [comment]
Fixed both issues. Poked around looking for optimizations tonight. Didn't have much luck, but made one small change.

My biggest worry was signal.GetType.BaseType and although BaseType is 3x slower than GetType, it's still blazing fast in the scheme of things. I'm guessing that is faster due to some (c#) caching, but I don't know if I could write a test that tests it without that layer. Suffice it to say it's plenty fast and I've moved along.

The rest is all delegate creation. There's not much we can do here, if you want to dynamically create delegates, you have to dynamically create delegates :) Delegate creation is slowest. That part will remain slow, sadly. 

Bottom line: Don't do this where performance is a concern! The performance is fine for all reasonable situations. Any situation requiring constant shuffling of views will use pooling, so you'll have a handle on performance anyways.

**Comment by WillzZz** [comment]
Updated description to escape the `<>` that was being ignored by markdown :P

**Comment by WillzZz** [comment]
This is going to tip for v1 release. Closing and reopening
